superset/models/dashboard.py

```diff
@@ -240,6 +251,29 @@ def data(self) -> Dict[str, Any]:
             "last_modified_time": self.changed_on.replace(microsecond=0).timestamp(),
         }
 
+    @cache.memoize(
+        # manually maintain cache key version
+        make_name=lambda fname: f"{fname}-v1",
+        timeout=config["DASHBOARD_CACHE_TIMEOUT"],
+        unless=lambda: not is_feature_enabled("DASHBOARD_CACHE"),
+    )
+    def full_data(self) -> Dict[str, Any]:
+        """Bootstrap data for rendering the dashboard page."""
+        slices = self.slices
+        datasource_slices = utils.indexed(slices, "datasource")
+        return {
```

Comment:
does any of this metadata rely on checking user access to the datasources? Double checking

Reply:
No it doesn't. All metadata is returned. Access checking is done in the [view](https://github.com/apache/incubator-superset/blob/19c63954a8432b8d0c049d02c4b8dae6764478f7/superset/views/core.py#L1622).